src/ol/renderer/canvas/ImageLayer.js

```diff
@@ -108,6 +108,7 @@ class CanvasImageLayerRenderer extends IntermediateCanvasRenderer {
 
     if (layerState.layer.getType() === LayerType.VECTOR) {
       renderedExtent = buffer(frameState.extent, layerState.layer.getImageDrawBuffer() * viewResolution);
+      frameState.extent = renderedExtent;
```

Comment:
Although it's tempting to do this, `frameState` must not be modified by layer renderers, because subsequent layers will then render with an incorrect extent. Instead, you'd have to store and use that buffered extent separately.